examples/SampleUtils.cpp

```diff
@@ -0,0 +1,192 @@
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "SampleUtils.h"
+
+#include "common/Assert.h"
+#include "common/Log.h"
+
+#include <webnn/webnn.h>
+#include <webnn/webnn_cpp.h>
+#include <webnn/webnn_proc.h>
+#include <webnn_native/WebnnNative.h>
+#if defined(_WIN32)
+#    include <crtdbg.h>
+#endif
+
+uint32_t product(const std::vector<int32_t>& dims) {
+    uint32_t prod = 1;
+    for (size_t i = 0; i < dims.size(); ++i)
+        prod *= dims[i];
+    return prod;
+}
+
+webnn::NeuralNetworkContext CreateCppNeuralNetworkContext() {
+    WebnnProcTable backendProcs = webnn_native::GetProcs();
+    webnnProcSetProcs(&backendProcs);
+    WebnnNeuralNetworkContext context = webnn_native::CreateNeuralNetworkContext();
+    if (context) {
+        return webnn::NeuralNetworkContext::Acquire(context);
+    }
+    return webnn::NeuralNetworkContext();
+}
+
+webnn::NamedInputs CreateCppNamedInputs() {
+    return webnn::NamedInputs::Acquire(webnn_native::CreateNamedInputs());
```

Comment:
I don't have all the context yet, but having to call webnn_native::Foo to create root objects is a bit weird. In Dawn we did this to bootstrap things but now the instance object can be created directly from wgpu:: and not dawn_native::. You might want to do something like this.

Reply:
good suggestion, fixed in the new commit https://github.com/webmachinelearning/webnn-native/pull/1/commits/0b4821ff5a7e51c5168aead07e86082aaf205728